example/weather/services/tester/run_tests.go

```diff
@@ -162,22 +201,50 @@ func (svc *Service) runTests(ctx context.Context, p *gentester.TesterPayload, te
 
 	// Run the tests that need to be run and add the results to the testCollection.Results array
 	if runSynchronously {
+		// RunSynchronously is used for test collections that need to be run one after in order
+		// to avoid single resource contention between tests if they are run in parallel. An
+		// example of this is tests that rely on the same cloud resource, such as a spreadsheet,
+		// as part of their test functionality.
+		//
+		// testName is passed to recoverFromTestPanic as a function so that, via a closure, its
+		// name can be set before the test is run but after the defer of recoverFromTestPanic is
+		// declared. This is done because the test name is not accessible from within the test
+		// function itself where it is set.
 		log.Infof(ctx, "RUNNING TESTS SYNCHRONOUSLY")
-		for n, test := range testsToRun {
-			log.Infof(ctx, "RUNNING TEST [%v]", n)
-			test(ctx, testCollection)
-		}
+		wg := sync.WaitGroup{}
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			testName := ""
+			testNameFunc := func() string {
+				return testName
+			}
+			defer recoverFromTestPanic(ctx, testNameFunc, testCollection)
+			for _, test := range testsToRun {
+				testName = getTestName(test)
```

Comment:
I am not sure I understand the need for `getTestName`. In general using reflection for a service at runtime isn't great (it's slow) and a bit of a code smell. At the very least if we were to use `getTestName` we could get rid of lines 218-221. But it would be much better to find a solution that does not require it. Is `testsToRun` not indexed by test name?

Reply:
@raphael It is, but I was having trouble getting the true test name in the case of wildcard filters. I'll take some time to see if I can't rework this. If I fail to do that I may need to keep it though so that the true test name is returned, especially when there is a panic so that the exact test that is panicking can be identified